src/reactor/cache.js

```diff
@@ -0,0 +1,201 @@
+import { List, Map, OrderedSet, Record } from 'immutable'
+
+export const CacheEntry = Record({
+  value: null,
+  storeStates: Map(),
+  dispatchId: null,
+})
+
+/*******************************************************************************
+ * interface PersistentCache {
+ *    has(item)
```

Comment:
Got it. I figured the reason was to keep the cache construct immutable, IMO would be simpler for developers to allow mutability and go with `get` + `put`